src/apis/software-service.ts

```diff
@@ -0,0 +1,109 @@
+import { DocumentClient } from 'aws-sdk/clients/dynamodb';
+import { v4 as uuidv4 } from 'uuid';
+import { SoftwareModel, Status } from './schemas/software-registry/software';
+
+const tableName = process.env.DYNAMODB_TABLE;
+
+/**
+ * Database service for software entries
+ */
+class SoftwareService {
+
+  /**
+   * Constructor
+   * @param docClient DynamoDB client
+   */
+  constructor(private readonly docClient: DocumentClient) {}
+
+  /**
+   * Creates a software entry
+   * @param software software entry
+   * @returns created software entry
+   */
+  public async createSoftware(software: Omit<SoftwareModel, 'id' | 'status' | 'createdAt' | 'lastUpdatedAt'>): Promise<SoftwareModel> {
```

Comment:
Is it actually necessary to use the Omit here, you could still set the id status etc as you are.
I'm not saying you are wrong, I just haven't seen this used often so would like to check in the reason for using it.